src/main/kotlin/com/wafflestudio/account/api/client/GithubClient.kt

```diff
@@ -0,0 +1,43 @@
+package com.wafflestudio.account.api.client
+
+import com.wafflestudio.account.api.domain.account.enum.SocialProvider
+import com.wafflestudio.account.api.interfaces.oauth2.GithubOAuth2UserResponse
+import com.wafflestudio.account.api.interfaces.oauth2.GithubOAuth2UserResponseBody
+import com.wafflestudio.account.api.interfaces.oauth2.OAuth2UserResponse
+import kotlinx.coroutines.reactor.awaitSingleOrNull
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.bodyToMono
+import reactor.core.publisher.Mono
+
+@Component("GITHUB")
+class GithubClient(
+    webClientHelper: WebClientHelper,
+    clientRegistrationRepository: ReactiveClientRegistrationRepository,
+) : OAuth2Client {
+
+    private val webClient = webClientHelper.buildWebClient()
+    private val clientRegistration = clientRegistrationRepository.findByRegistrationId(
+        SocialProvider.GITHUB.value
+    ).block()!!
+
+    override suspend fun getMe(
+        accessToken: String
+    ): OAuth2UserResponse? {
+        return webClient
+            .get()
+            .uri(clientRegistration.providerDetails.userInfoEndpoint.uri)
+            .headers {
+                it.setBearerAuth(accessToken)
+            }
+            .retrieve()
+            .bodyToMono<GithubOAuth2UserResponseBody>()
+            .onErrorResume {
+                WebClientHelper.logger.error(it.message, it)
+                Mono.empty()
+            }
+            .flatMap {
+                Mono.just(GithubOAuth2UserResponse(it.email, it.socialId.toString()))
+            }.awaitSingleOrNull()
```

Comment:
추가적으로 아주 간단한 코멘트 달자면

```suggestion
            .map {
                GithubOAuth2UserResponse(it.email, it.socialId.toString())
            }.awaitSingleOrNull()
```

여기서 flatMap은 map으로 변경하는게 나을 것 같아요
map은 스트림 내에서 데이터만 조금 변경하는 거고
flatMap은 아예 스트림 자체를 벗겨내고 새로 만들기 때문에
성능차이가 큰건아니지만 조금 있어요

flatMap을 사용하는 경우는 새로운 nonblocking IO 작업이 필요할 때 사용합니다.

다른 곳도 마찬가지로 변경하면 좋을 것 같아요

Reply:
^ 이름이 flatMap인 이유

Reply:
반영했습니다! 좋은 지적과 이유 설명 감사해요~